### PR TITLE
chore(medusa): cleanup medusa package

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
- Remove a bunch of things from medusa package (not entirely cleaned up but we can do a second iteration for that)
- Remove medusa-js
- Remove medusa-react
- Fix tests
- Remove some v1 specific thing

Another iteration will be necessary to remove the left over and the piece of code that depends on the v2 flags, we should only have v2 specific left after the next iterations

### CODE REVIEW

File: .github/workflows/test-cli-with-database.yml [hunk 1]
```diff
@@ -35,56 +35,56 @@ jobs:
...
-
-      ###########################  Test medusa start  ################################
-
-      - name: Starting medusa
-        run: medusa start &
-        working-directory: ../cli-test
-
-      - name: Testing server
-        uses: ./.github/actions/test-server
+    #steps:
```
Comment (olivermrbl) [inline .github/workflows/test-cli-with-database.yml]: thought: I know you are not done yet, but maybe we can port this to test a V2 setup?

That would require supporting a V2 flag similar to `create-medusa-app` in the `new` command. Maybe @shahednasser can help with this, if we want to have this.

Comment (adrien2p) [reply .github/workflows/test-cli-with-database.yml]: yes sure, @shahednasser would you be able to jump on the pr while I continue?

Comment (shahednasser) [reply .github/workflows/test-cli-with-database.yml]: Just to be clear, we want to add to the `new` command a `--v2` option that uses the same starter branch as in `create-medusa-app`, correct?

Comment (olivermrbl) [reply .github/workflows/test-cli-with-database.yml]: Yes, the exact same behavior and option as `create-medusa-app`, but in `new.ts`

Comment (shahednasser) [reply .github/workflows/test-cli-with-database.yml]: Created PR #7211

Comment (olivermrbl) [reply .github/workflows/test-cli-with-database.yml]: Epic, thanks @shahednasser 🙏

**Comment by adrien2p** [comment]
i ll put this pr on pause until other issues are fixed

File: .github/workflows/test-cli-with-database.yml (hunk 1 above)
Comment (olivermrbl) [reply .github/workflows/test-cli-with-database.yml]: #7211 has been merged, so we can update the action :)

**Review by sradevski — APPROVED** [review]

**Comment by adrien2p** [comment]
@olivermrbl would have a bit of time to review this one please ?

**Comment by adrien2p** [comment]
we might need to make the non running actions non required for a moment?

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Review by olivermrbl — APPROVED** [review]

**Comment by olivermrbl** [comment]
@kasperkristensen @fPolic FYI, we are removing medusa-react and medusa-js temporarily due to some workspace hoisting issues.  We can recreate those packages when we figure out what we want to do with the client + hooks in `dashboard`.

**Comment by adrien2p** [comment]
about the require actions, will you be able to rm the api, plugins and test cli with db for now? @olivermrbl

**Comment by olivermrbl** [comment]
I can force merge, if everyone is cool with that

**Comment by adrien2p** [comment]
for sure @olivermrbl

**Pull request merged**